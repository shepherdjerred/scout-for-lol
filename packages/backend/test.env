VERSION=test
GIT_SHA=test
ENVIRONMENT=dev
DISCORD_TOKEN=test
APPLICATION_ID=test
RIOT_API_TOKEN=test
DATABASE_URL=file:./db.sqlite

# S3 Storage Configuration (optional - leave commented for testing)
# S3_BUCKET=scout-lol-matches-test
# AWS_REGION=us-east-1
# AWS_ACCESS_KEY_ID=your-access-key
# AWS_SECRET_ACCESS_KEY=your-secret-key
# S3_ENDPOINT=https://s3.amazonaws.com
# S3_FORCE_PATH_STYLE=false
