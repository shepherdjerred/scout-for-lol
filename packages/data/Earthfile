VERSION 0.8
FROM ../../+deno

src:
  COPY src .
  SAVE ARTIFACT *

deps:
  WORKDIR /workspace/packages/data

  # download and cache dependencies
  CACHE --persist $DENO_DIR
  COPY --dir deno* src .
  RUN deno cache src/index.ts

check:
  FROM +deps
  CACHE $DENO_DIR
  RUN deno check src/index.ts
  RUN deno lint
  # RUN deno test -A --unstable
