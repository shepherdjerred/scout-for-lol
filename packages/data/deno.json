{
  "tasks": {
    "check": "deno check src/index.ts",
    "check:watch": "deno run -A https://deno.land/x/denon/denon.ts check",
    "test:update": "deno test --allow-all -- --update"
  },
  "imports": {
    "@std/testing": "jsr:@std/testing@1.0.6",
    "remeda": "https://esm.sh/remeda@2.17.4",
    "ts-pattern": "https://esm.sh/ts-pattern@5.5.0",
    "zod": "https://esm.sh/zod@3.24.1"
  }
}
