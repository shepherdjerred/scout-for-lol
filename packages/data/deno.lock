{
  "version": "4",
  "specifiers": {
    "jsr:@std/assert@1.0.10": "1.0.10",
    "jsr:@std/assert@^1.0.10": "1.0.10",
    "jsr:@std/async@^1.0.9": "1.0.9",
    "jsr:@std/data-structures@^1.0.5": "1.0.5",
    "jsr:@std/fs@^1.0.8": "1.0.9",
    "jsr:@std/internal@^1.0.5": "1.0.5",
    "jsr:@std/path@^1.0.8": "1.0.8",
    "jsr:@std/testing@1.0.8": "1.0.8"
  },
  "jsr": {
    "@std/assert@1.0.10": {
      "integrity": "59b5cbac5bd55459a19045d95cc7c2ff787b4f8527c0dd195078ff6f9481fbb3",
      "dependencies": [
        "jsr:@std/internal"
      ]
    },
    "@std/async@1.0.9": {
      "integrity": "c6472fd0623b3f3daae023cdf7ca5535e1b721dfbf376562c0c12b3fb4867f91"
    },
    "@std/data-structures@1.0.5": {
      "integrity": "86dd5c2d6365a6ad56e9103b3a877c5d5ade238c68ae463acfe0eb0e56d40783"
    },
    "@std/fs@1.0.9": {
      "integrity": "3eef7e3ed3d317b29432c7dcb3b20122820dbc574263f721cb0248ad91bad890",
      "dependencies": [
        "jsr:@std/path"
      ]
    },
    "@std/internal@1.0.5": {
      "integrity": "54a546004f769c1ac9e025abd15a76b6671ddc9687e2313b67376125650dc7ba"
    },
    "@std/path@1.0.8": {
      "integrity": "548fa456bb6a04d3c1a1e7477986b6cffbce95102d0bb447c67c4ee70e0364be"
    },
    "@std/testing@1.0.8": {
      "integrity": "ceef535808fb7568e91b0f8263599bd29b1c5603ffb0377227f00a8ca9fe42a2",
      "dependencies": [
        "jsr:@std/assert@^1.0.10",
        "jsr:@std/async",
        "jsr:@std/data-structures",
        "jsr:@std/fs",
        "jsr:@std/internal",
        "jsr:@std/path"
      ]
    }
  },
  "remote": {
    "https://esm.sh/remeda@2.19.0": "718d24e624c5c1e0900223dd88ea9a7fd8087885a1ef648340241a0561a22b70",
    "https://esm.sh/ts-pattern@5.6.0": "7abf888444efe5f68539d30842709badfc9495f86fe61a4755a4cfa004b6f834",
    "https://esm.sh/v135/remeda@2.19.0/denonext/remeda.mjs": "8473eca3903d8517b1d9f7ebfcb019b4aaf2d1e2d082a3209142a93cd07cca3a",
    "https://esm.sh/v135/ts-pattern@5.6.0/denonext/ts-pattern.mjs": "518a6b3f30ecd5cae77e1293e78da1ccba1387592458d9587e25dc43277909d3",
    "https://esm.sh/v135/type-fest@4.30.2/denonext/source/simplify.d.js": "6bb4fd7ea1ef11ee180be67356c5f42eb9812d243c21257d34f2c14effb5d665",
    "https://esm.sh/v135/type-fest@4.30.2/denonext/source/union-to-intersection.d.js": "131563cf651c20060eb48f1f7cb9f3312cbe5c56c2ff858dc693dec52d26d941",
    "https://esm.sh/v135/type-fest@4.30.2/source/simplify.d": "628458b10e498e2d5218acd3e979a897089260f33a54a19be97359eee23e4e0c",
    "https://esm.sh/v135/type-fest@4.30.2/source/union-to-intersection.d": "5c9902261f13940fa13f42704bc9ddb138e93281419f77342c0b70e9fd5c957b",
    "https://esm.sh/v135/zod@3.24.1/denonext/zod.mjs": "267cf807d908327192aa402850ce2cf1c199c02b097f4c521272a42538dad53a",
    "https://esm.sh/zod@3.24.1": "c6bc4bbea04826432b2b4e66b77316c0fa90d1132e879dd0193c90f9114436c4"
  },
  "workspace": {
    "dependencies": [
      "jsr:@std/assert@1.0.10",
      "jsr:@std/testing@1.0.8"
    ]
  }
}
