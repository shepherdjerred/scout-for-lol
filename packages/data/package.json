{
  "name": "@scout-for-lol/data",
  "version": "0.1.0",
  "type": "module",
  "main": "src/index.ts",
  "scripts": {
    "test": "bun test",
    "lint": "bunx eslint src --ext .ts",
    "format": "bunx prettier --check src",
    "format:write": "bunx prettier --write src",
    "typecheck": "bunx tsc --noEmit"
  },
  "dependencies": {
    "remeda": "^2.21.2",
    "ts-pattern": "^5.7.0",
    "zod": "^3.24.2"
  },
  "devDependencies": {
    "typescript": "^5.0.0",
    "@typescript-eslint/eslint-plugin": "^6.0.0",
    "@typescript-eslint/parser": "^6.0.0",
    "eslint": "^8.0.0",
    "prettier": "^3.0.0",
    "bun-types": "latest"
  }
}
