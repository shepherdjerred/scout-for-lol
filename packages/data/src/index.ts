export * from "./model/index.ts";
