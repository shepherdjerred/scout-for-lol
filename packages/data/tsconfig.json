{
  "compilerOptions": {
    "target": "ES2022",
    "module": "ES2022",
    "lib": ["ES2022"],
    "moduleResolution": "node",
    "strict": true,
    "esModuleInterop": true,
    "skipLibCheck": true,
    "forceConsistentCasingInFileNames": true,
    "resolveJsonModule": true,
    "allowJs": true,
    "types": ["bun-types"],
    "allowUnreachableCode": false,
    "allowUnusedLabels": false,
    "exactOptionalPropertyTypes": true,
    "noFallthroughCasesInSwitch": true,
    "noImplicitOverride": true,
    "noImplicitReturns": true,
    "noPropertyAccessFromIndexSignature": true,
    "noUncheckedIndexedAccess": true,
    "noUnusedLocals": true,
    "noUnusedParameters": true,
    "useUnknownInCatchVariables": true
  },
  "include": ["src/**/*"],
  "exclude": ["node_modules", "vendor"]
}
