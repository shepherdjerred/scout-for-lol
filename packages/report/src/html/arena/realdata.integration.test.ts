import { test, expect } from "bun:test";
import { writeFileSync } from "fs";
import { createHash } from "crypto";
import { arenaMatchToSvg } from "./index.tsx";
import { dirname, join } from "path";
import { fileURLToPath } from "url";
import { ArenaMatchSchema } from "@scout-for-lol/data";
import { svgToPng } from "../index.tsx";

const currentDir = dirname(fileURLToPath(import.meta.url));

function hashSvg(svg: string): string {
  return createHash("sha256").update(svg).digest("hex");
}

const RAW_FILE_PATHS = [
  join(currentDir, "testdata/1.json"),
  join(currentDir, "testdata/2.json"),
];

for (const path of RAW_FILE_PATHS) {
  const fileName = path.split("/").pop();
  if (!fileName) {
    throw new Error("fileName is undefined");
  }
  const testName = `arena real data renders image for ${fileName}`;
  test(testName, async () => {
    const match = (await Bun.file(path).json()) as unknown;
    const svg = await arenaMatchToSvg(ArenaMatchSchema.parse(match));
    const png = svgToPng(svg);
    expect(svg.length).toBeGreaterThan(1024); // basic sanity check
    const fileName =
      path.split("/").pop()?.replace(".json", ".png") ?? "arena_real.png";
    writeFileSync(new URL(`__snapshots__/${fileName}`, import.meta.url), png);

    // Hash the SVG for snapshot comparison instead of storing the full content
    const svgHash = hashSvg(svg);
    expect(svgHash).toMatchSnapshot();
  });
}
