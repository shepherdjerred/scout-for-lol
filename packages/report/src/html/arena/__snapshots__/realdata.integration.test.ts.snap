// Bun Snapshot v1, https://bun.sh/docs/test/snapshots

exports[`arena real data renders image for 1.json 1`] = `"f703c6841b446165ab647a7dadca9cb63b63c433f43a6459b4ed8eace32fb9ad"`;

exports[`arena real data renders image for 2.json 1`] = `"ae74dc6ba74df5ffa3f3b5ac378ce3a4a5843fe32e322776013cb4ada42cd178"`;
